webvr-migration.md

```diff
@@ -0,0 +1,298 @@
+
+**WebVR**
+```js
+function onFrame (t) {
+  // Queue a request for the next frame to keep the animation loop going.
+  vrDisplay.requestAnimationFrame(onFrame);
+
+  // Loop through all gamepads and identify the ones that are associated with
+  // the vrDisplay.
+  let gamepads = navigator.getGamepads();
+  for (let i = 0; i < gamepads.length; ++i) {
+    let gamepad = gamepads[i];
+    // The array may contain undefined gamepads, so check for that as
+    // well as a non-null pose.
+    if (gamepad && gamepad.displayId && gamepad.pose) {
+      scene.showControllerAtTransform(gamepad.pose.position, gamepad.pose.orientation, gamepad.hand);
+    }
+  }
+
+  // Handle rendering as shown above...
+}
+```
+
+**WebXR**
+```js
+function onFrame(t, frame) {
+  // Queue a request for the next frame to keep the animation loop going.
+  xrSession.requestAnimationFrame(onXRFrame);
+
+  // Loop through all input sources.
+  for (let inputSource of xrSession.inputSources) {
+    // Show the input source if it has a grip space
+    if (inputSource.gripSpace) {
+      let inputPose = frame.getPose(inputSource.gripSpace, xrReferenceSpace);
+      scene.showControllerAtTransform(inputPose.position, inputPose.orientation, inputSource.handedness);
+    }
+  }
+
+  // Handle rendering as shown above...
+);
+```
+
+Ending VR presentation
+----------------------
+Both WebVR and WebXR may have their presentation of VR content ended by the UA at any time.
+
+WebVR applications may explicitly end the presentation of VR content by calling `vrDisplay.exitPresent()`. A `vrdisplaypresentchange` event is fired on the `Window` object when presentation is started or ended by either the application or the UA. The presentation state is determined by checking the `vrDisplay.isPresenting` boolean attribute.
+
+WebXR applications may explicitly end the presentation of VR content by calling [`xrSession.end()`](https://immersive-web.github.io/webxr/#dom-xrsession-end), at which point the [`XRSession`](https://immersive-web.github.io/webxr/#xrsession-interface) object becomes unusable. An [`end`](https://immersive-web.github.io/webxr/#eventdef-xrsession-end) event is fired on the [`XRSession`](https://immersive-web.github.io/webxr/#xrsession-interface) when it is ended by either the application or UA.
+
+**WebVR**
+```js
+window.addEventListener("vrdisplaypresentchange", () => {
+  if (!vrDisplay.isPresenting) {
+    // VR presentation has ended. Do any necessary cleanup.
+  }
+});
+```
+
+**WebXR**
+```js
+xrSession.addEventListener("end", () => {
+  // VR presentation has ended. Do any necessary cleanup.
+});
+```
+
+Misc
+----
+WebXR currently has no equivalent for the `vrdisplayactivate`, `vrdisplaydeactivate`, `vrdisplaypointerrestricted`, and `vrdisplaypointerunrestricted` events.
```

Comment:
Also no equivalent of https://immersive-web.github.io/webvr/spec/1.1/#dom-vrdisplayeventreason-navigation